app/assets/javascripts/components/timeline/week.jsx

```diff
@@ -68,21 +71,35 @@ const Week = createReactClass({
     let style;
     const dateCalc = new DateCalculator(this.props.timeline_start, this.props.timeline_end, this.props.index, { zeroIndexed: false });
 
-    let weekDates;
+    let weekDatesContent;
     if (this.props.meetings) {
-      weekDates = (
-        <span className="week__week-dates pull-right">
-          {dateCalc.start()} - {dateCalc.end()} {this.props.meetings}
-        </span>
-      );
+      weekDatesContent = `${dateCalc.start()} - ${dateCalc.end()} ${this.props.meetings}`;
     } else {
-      weekDates = (
-        <span className="week__week-dates pull-right">
-          Week of {dateCalc.start()} — AFTER TIMELINE END DATE!
-        </span>
-      );
+      weekDatesContent = `Week of ${dateCalc.start()} — AFTER TIMELINE END DATE!`;
     }
+    const weekDates = (
+      <div className="week__week-dates">
+        {weekDatesContent}
+      </div>
+    );
 
+    let weekTitleContent;
+    if (this.props.week.title) {
+      weekTitleContent = this.props.week.title;
+    } else {
+      weekTitleContent = I18n.t('timeline.week_number', { number: this.weekNumber() });
+    }
+    const weekId = this.props.week.id;
+    const weekTitle = this.props.editableTitles ? (
+      <input
+        className="week-index week-title-input"
+        defaultValue={weekTitleContent}
+        maxLength={20}
+        onChange={text => this.props.updateTitle(weekId, event.target.value, text)}
```

Comment:
This is giving me an error. Should be `event => this.props.updateTitle(weekId, event.target.value)`?

Reply:
This error just happens in Firefox, it looks like.